### PR TITLE
Add multiple selection for about:history

### LINKED ISSUE #3949: Add multi-select deletion and drag and drop in about:history
**Expected behavior:** Please provide ways to delete entries on about:history selectively. For example,

- checkboxes
- multiple selection with <kbd>alt</kbd>/<kbd>command</kbd> key

### PR DESCRIPTION
**Pull request by darkdh**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

fix #3949

Auditors: @bsclifton, @bbondy

Test Plan:
For click behavior
---
1. First click should select a entry
2. All the following entries should be selectd with cmd/ctrl + click
3. Another click without cmd/ctrl should cancel selection

For shift click behavior
---
1. Click one entry
2. Shift + click another entry in the same table
3. The entry between these two will also be selected

For shift click + cmd/control click
---
1. Click one entry as start point
2. command/control click other entries between start point and end point in next step
3. Shift + click an entry as end point
4. The entry between start and end point should be selected

For multi selector
---
1. Open about:history
2. Multi-select entries
3. Test "Open Links in New Tabs"
4. Test "Open Links in New Private Tabs"
5. Test "Open Links in New Session Tabs"
6. Test "Delete History Entries"

### CODE REVIEW

**Comment by bsclifton** [comment]
Great job :smile: 

One concern I have is for the selected items. @bradleyrichter, what do you think about the colors?
screen shot 2016-10-07 at 2 09 30 pm

The dark orange is multi-selected, the light orange is hovered. If you hover over a multi-select, it looks like this:
screen shot 2016-10-07 at 2 11 14 pm

It feels good- but the concern I had is readability. We may choose to change the font color (white or lighter color) to make selected items easier to read

File: app/common/lib/platformUtil.js [hunk 1]
```diff
@@ -27,9 +27,11 @@ module.exports.getPlatformStyles = () => {
...
 module.exports.isDarwin = () => {
-  return os.platform() === 'darwin'
+  return os.platform() === 'darwin' ||
+    navigator.platform === 'MacIntel'
 }
 
 module.exports.isWindows = () => {
-  return os.platform() === 'win32'
+  return os.platform() === 'win32' ||
+    navigator.platform === 'Win32'
```
Comment (bsclifton) [inline app/common/lib/platformUtil.js]: thanks for adding these checks in :smile:

**Comment by bsclifton** [comment]
Reviewed the code in detail, changes look great :+1:

**Comment by darkdh** [comment]
Enhanced readability as 
screen shot 2016-10-08 at 16 52 22
in https://github.com/brave/browser-laptop/pull/4607/commits/dcc8e495c6335d881db1613e04cb22c6e4e75176

**Comment by bbondy** [comment]
Could you also select Shift + select for range selections?

**Comment by darkdh** [comment]
Working on it

**Comment by bsclifton** [comment]
++ looks great :smile: Thanks for spending the time to make the tests too

**Pull request merged**